File: programmers/Level 2/remove_pair.cpp
/*
programmers Level 2: 짝지어 제거하기
date: 2022-06-01
*/
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int solution(string s)
{
    int answer = -1;
    stack<char> st;
    
    for(auto i : s){
        if(st.empty() || st.top() != i) st.push(i);
        else st.pop();
    }
    
    if(st.empty()) answer = 1;
    else answer = 0;
    
    return answer;
}

File: programmers/Level 2/remove_pair.py
# programmers Level 2: 짝지어 제거하기
# date: 2022-06-10
def solution(s):
    answer = -1
    list = []
    
    for i in s:
        if len(list) == 0 or list[-1] != i:
            list.append(i)
        elif list[-1] == i:
            list.pop()
    
    if len(list) > 0: answer = 0
    else: answer = 1
    
    return answer